memory_profiler.py

```diff
@@ -48,6 +52,15 @@ def unicode(x, *args): return str(x)
 except ImportError:
     has_psutil = False
 
+try:
+    import tracemalloc
+    has_tracemalloc = True
+except ImportError:
+    has_tracemalloc = False
+
+_backend_chosen = False
+_backend = 'psutil'
+
```

Comment:
Do you need two global variables? I think you can do with only one (possibly initialized to None).